File: Assets/Scripts/CSharp Scripts/ElevatorDoorScript.cs
using UnityEngine;
using System.Collections;

public class ElevatorDoorScript : MonoBehaviour {

	public GameObject ConnectedPlate1;
	public GameObject ConnectedPlate2;
	public GameObject key;

	private PlateScript plateScript1;
	private PlateScript plateScript2;

	private bool needsKey;
	private bool needsBox;
	private bool needsBoxes;
	private bool openDoor;

	public Transform player;


	// Use this for initialization
	void Start () {
		if (ConnectedPlate1) 
		{
			plateScript1 = ConnectedPlate1.GetComponent<PlateScript> ();
			needsBox = true;
			if (ConnectedPlate2) 
			{
				plateScript2 = ConnectedPlate2.GetComponent<PlateScript> ();
				needsBoxes = true;
			}
		}
		needsKey = (key != null);
	}
	
	// Update is called once per frame
	void Update () {
		if (needsBox) {
			if (needsBoxes) {
				if (plateScript1.activated && plateScript2.activated) {
					openDoor = true;
				} else {
					openDoor = false;
				}
			} else {
				if (plateScript1.activated) {
					openDoor = true;
				} else {
					openDoor = false;
				}
			}
		} else if (needsKey) {
			if (key == null) {
				openDoor = true;
			} else {
				openDoor = false;
			}
		} else if (Close (player, transform, 4)) 
		{
			openDoor = true;
		}



		BoxCollider box = GetComponent<BoxCollider> ();
		Vector3 scale = transform.localScale;
		Vector3 boxScale = box.size;

		if (openDoor) 
		{
			if (transform.localScale.x > 0) 
			{
				scale.x -= Time.deltaTime * .4f;
				transform.localScale = scale;
			}
			if (box.size.x > 0) 
			{
				boxScale.x -= Time.deltaTime * .4f;
				box.size = boxScale;
			}
		} else 
		{
			if (transform.localScale.x < 1) 
			{
				scale.x += Time.deltaTime * .4f;
				transform.localScale = scale;
			}
			if (box.size.x < 1)
			{
				boxScale.x += Time.deltaTime * .4f;
				box.size = boxScale;
			}
		}
	}

	bool Close (Transform one, Transform two, int length)
	{
		return (Mathf.Abs(one.position.x - two.position.x) +  Mathf.Abs(one.position.z - two.position.z)) < length;
	}
}

File: Assets/Scripts/New Scripts/ElevatorDoorScript.js
#pragma strict
var connectedPlate1: GameObject;
var connectedPlate2: GameObject;
var plateScript1: NewPlateScript;
var plateScript2: NewPlateScript;
var key: GameObject;
var delay = 0;
private var needsKey: boolean;
private var needsBox: boolean;
private var needsBoxes: boolean;
var chara:Transform;
private var openDoor: boolean;


function Close (one: Transform, two: Transform, length: int){
	return (Mathf.Abs(one.position.x - two.position.x) +  Mathf.Abs(one.position.z - two.position.z)) < length;
}

function Start () {
	if (connectedPlate1)
	{
		plateScript1 = connectedPlate1.GetComponent(NewPlateScript);
		needsBox = true;
		if (connectedPlate2)
		{
			plateScript2 = connectedPlate2.GetComponent(NewPlateScript);
			needsBoxes = true;
		}
	}
	needsKey = (key != null);

}

function Update () {
	var boxCollider = GetComponent(BoxCollider) as BoxCollider;
	if(needsBox){

		if (needsBoxes)
		{
			if (plateScript1.activated && plateScript2.activated)
			{
				openDoor = true;
			} else
			{
				openDoor = false;
			}

		} else
		{
			if (plateScript1.activated)
			{
				openDoor = true;

			} else
			{
				openDoor = false;
			}
		}
	} else if (needsKey)
	{

	} else if (Close(chara, transform, 3))
	{
		openDoor = true;
	}


	if(openDoor)
	{
		if(transform.localScale.x > 0)
		{
			transform.localScale.x -= Time.deltaTime * .4;
		}
		if(boxCollider.size.x > 0)
		{
			boxCollider.size.x -= Time.deltaTime * .4;
		}
	}
}